internal/pkg/agent/application/upgrade/watcher.go

```diff
@@ -133,12 +134,11 @@ LOOP:
 			// block on connection, don't retry connection, and fail on temp dial errors
 			// always a local connection it should connect quickly so the timeout is only 1 second
 			connectCtx, connectCancel := context.WithTimeout(ctx, 1*time.Second)
-			//nolint:staticcheck // requires changing client signature
 			err := ch.agentClient.Connect(connectCtx, grpc.WithBlock(), grpc.WithDisableRetry(), grpc.FailOnNonTempDialError(true))
```

Comment:
As a follow up to the linter message, seems like `WithBlock` and `FailOnNonTempDialError` are considered an [anti-pattern](https://github.com/grpc/grpc-go/blob/aae9e64/Documentation/anti-patterns.md).

Following up on the linter message, it appears that WithBlock and FailOnNonTempDialError are considered an [anti-pattern](https://github.com/grpc/grpc-go/blob/aae9e64/Documentation/anti-patterns.md).

It’s not entirely clear why these are used. They might be remnants of old code or included to work around an error or limitation. Could they be removed?